Support Systems without GLOB_BRACE (e.g. musl)

BjoernAkAManf opened the pull request:
Some Systems (e.g. alpine) do not support GLOB_BRACE directly. This will cause dependend implementation to fail as GLOB_BRACE will not be defined. This is fixed by using Stdlib from the Zend Framework, bundled with the current phpBB distribution. Systems with direct support of GLOB_BRACE should not be affected by this, as it will automatically be used if available.

Without this change affected systems might run into corruption of the autoloader used by the acp until manually deleting the cache inside the phpbb installation.

Since i haven't used phpbb in a long time, im not sure this is everything that is needed. Applying this solution can also be a temporary workaround for Version 1.1, as no major changes have been made. This should be every occasion of glob. I did not however add zen lib as a dependency, because it is already provided by phpBB and i didn't want to touch the build files, if required.

BjoernAkAManf committed:
Support Systems without GLOB_BRACE (e.g. musl)

Some Systems (e.g. alpine) do not support GLOB_BRACE directly. This will cause dependend implementation to fail as GLOB_BRACE will not be defined. This is fixed by using Stdlib from the Zend Framework, bundled with the current phpBB distribution. Systems with direct support of GLOB_BRACE should not be affected by this, as it will automatically be used if available.

Without this change affected systems might run into corruption of the autoloader used by the acp until manually deleting the cache inside the phpbb installation.

BjoernAkAManf commented:
Just to be sure: I disagree with the issues codeclimate reports. Since this is effectively a system call with a fallback.

stevotvr commented:
Would it be better to glob everything and filter the files using string functions? I don't know if we're meant to rely on the availability of the bundled libraries.

BjoernAkAManf commented:
I've read some posts, that claim a custom vendor folder might be used to include your own composer dependencies. This would allow us to bundle stdlib ourselves. But i have yet to reproduce that locally.

stevotvr commented:
It would be trivial to remove the braces and use string functions instead. Unless there are other issues that the library addresses, I would rather avoid using it.

BjoernAkAManf committed:
Replaced GLOB_BRACE with custom Implementation

Replaced glob brace implementation with simple readdir implementation.
Reverted back to old glob function calls where applicable.

BjoernAkAManf commented:
I've added an additional proposal. This should fix your concerns.

stevotvr commented on operator/image.php:
```diff
@@ -41,8 +41,7 @@ class image extends operator implements image_interface
 	 * @param \phpbb\filesystem\filesystem_interface $filesystem
-	 * @param string                                 $img_path      The path to the custom images
-	 */
+	 * @param string                                 $img_path      The path to the custom images	 */
```
Missing line break

stevotvr commented on operator/image.php:
```diff
@@ -90,23 +89,35 @@ public function get_images()
 		$images = array();
 
-		foreach (glob($this->img_path . '*{-x1.{gif,png,jpg,jpeg,GIF,PNG,JPG,JPEG},.{svg,SVG}}', GLOB_BRACE) as $file)
+		$handle = opendir($this->img_path);
```
Should there be a closedir somewhere?

stevotvr commented:
That looks like it would work. I added a few comments to the code.

BjoernAkAManf commented:
i did push it. Githubs web client is just literally annoying and i miss my cli.

BjoernAkAManf committed:
Fix Code Style

Signed-off-by: Björn Heinrichs <manfie@protonmail.ch>

BjoernAkAManf commented:
Fixed it using cli.
